discovery_engine_core/core/src/stack/ops/breaking.rs

```diff
@@ -60,16 +64,23 @@ impl Ops for BreakingNews {
             let mut articles = Vec::new();
             let mut errors = Vec::new();
 
+            let mut requests = FuturesUnordered::new();
             for market in markets.read().await.iter() {
-                let query = HeadlinesQuery {
-                    market,
-                    page_size: self.page_size,
-                };
-                match self.client.headlines(&query).await {
-                    Ok(batch) => articles.extend(batch),
-                    Err(err) => errors.push(err),
+                let request =
+                    spawn_headlines_request(self.client.clone(), market.clone(), self.page_size);
+                requests.push(request);
+            }
```

Comment:
the `requests` could be collected directly in a more functional style (i unfortunately can't mark it as a suggestion as the diff contains deleted lines). similarly for the personalized news query.
```rust
            let mut requests = markets
                .read()
                .await
                .iter()
                .cloned()
                .map(|market| spawn_headlines_request(self.client.clone(), market, self.page_size))
                .collect::<FuturesUnordered<_>>();
```